b.c

```diff
@@ -295,30 +295,30 @@ void freetr(Node *p)	/* free parse tree */
 /* in the parsing of regular expressions, metacharacters like . have */
 /* to be seen literally;  \056 is not a metacharacter. */
 
-int hexstr(uschar **pp)	/* find and eval hex string at pp, return new p */
+int hexstr(const uschar **pp)	/* find and eval hex string at pp, return new p */
 {			/* only pick up one 8-bit byte (2 chars) */
-	uschar *p;
+	const uschar *p;
 	int n = 0;
 	int i;
 
-	for (i = 0, p = (uschar *) *pp; i < 2 && isxdigit(*p); i++, p++) {
+	for (i = 0, p = (const uschar *) *pp; i < 2 && isxdigit(*p); i++, p++) {
```

Comment:
Why is this cast needed?

Reply:
This is only in the BSD branch though. Which brings the question: why have a separate branch? Perhaps we can enable the extra features with ifdefs? It is difficult to do cleanups otherwise.

Reply:
Ah, no I am wrong. But the question is still relevant.